### PR TITLE
Documentation Refresh

### PR DESCRIPTION
**Pull request by seanmalloy**
This pull requests cleans up the README.md file, creates a user guide, and creates a contributor guide. The user and contributor guides could use some additional improvements, but I believe in their current state will provide value to end users.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,9 +9,9 @@ Scheduling in Kubernetes is the process of binding pending pods to nodes, and is
 a component of Kubernetes called kube-scheduler. The scheduler's decisions, whether or where a
 pod can or can not be scheduled, are guided by its configurable policy which comprises of set of
 rules, called predicates and priorities. The scheduler's decisions are influenced by its view of
-a Kubernetes cluster at that point of time when a new pod appears first time for scheduling.
-As Kubernetes clusters are very dynamic and their state change over time, there may be desired
-to move already running pods to some other nodes for various reasons:
+a Kubernetes cluster at that point of time when a new pod appears for scheduling.
+As Kubernetes clusters are very dynamic and their state changes over time, there may be desire
+to move already running pods to some other nodes for various reasons.
```
Comment (ingvagabund) [inline README.md]: `:` is still more suitable here

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ being able to be run multiple times without needing user intervention.
 The descheduler pod is run as a critical pod to avoid being evicted by itself,
 or by the kubelet due to an eviction event. Since critical pods are created in the
 `kube-system` namespace, the descheduler job and its pod will also be created
-in `kube-system` namespace.
+in the `kube-system` namespace.
```
Comment (ingvagabund) [inline README.md]: nit: `in the same namespace` sounds a bit better since `kube-system` is mentioned only once

File: README.md [hunk 3]
```diff
@@ -84,20 +84,21 @@ $ kubectl create -f kubernetes/cronjob.yaml
...
 This strategy makes sure that there is only one pod associated with a Replica Set (RS),
-Replication Controller (RC), Deployment, or Job running on same node. If there are more,
+Replication Controller (RC), Deployment, or Job running on the same node. If there are more,
 those duplicate pods are evicted for better spreading of pods in a cluster. This issue could happen
 if some nodes went down due to whatever reasons, and pods on them were moved to other nodes leading to
-more than one pod associated with RS or RC, for example, running on same node. Once the failed nodes
+more than one pod associated with a RS or RC, for example, running on the same node. Once the failed nodes
 are ready again, this strategy could be enabled to evict those duplicate pods. Currently, there are no
-parameters associated with this strategy. To disable this strategy, the policy should look like:
+parameters associated with this strategy. To disable this strategy, the policy should look like this.
```
Comment (ingvagabund) [inline README.md]: `:` is still more suitable here

File: README.md [hunk 4]
```diff
@@ -116,15 +117,15 @@ parameters of this strategy are configured under `nodeResourceUtilizationThresho
 The under utilization of nodes is determined by a configurable threshold, `thresholds`. The threshold
```
Comment (ingvagabund) [inline README.md]: `,` can be removed

File: README.md [hunk 5]
```diff
@@ -116,15 +117,15 @@ parameters of this strategy are configured under `nodeResourceUtilizationThresho
...
+Currently, pods request resource requirements are considered for computing node resource utilization.
 
 There is another configurable threshold, `targetThresholds`, that is used to compute those potential nodes
 from where pods could be evicted. Any node, between the thresholds, `thresholds` and `targetThresholds` is
 considered appropriately utilized and is not considered for eviction. The threshold, `targetThresholds`,
 can be configured for cpu, memory, and number of pods too in terms of percentage.
 
 These thresholds, `thresholds` and `targetThresholds`, could be tuned as per your cluster requirements.
-An example of the policy for this strategy would look like:
+Here is an example of a policy for this strategy.
```
Comment (ingvagabund) [inline README.md]: `:` is still more suitable here

File: README.md [hunk 6]
```diff
@@ -144,14 +145,14 @@ strategies:
...
 
 ### RemovePodsViolatingInterPodAntiAffinity
 
-This strategy makes sure that pods violating interpod anti-affinity are removed from nodes. For example, if there is podA on node and podB and podC(running on same node) have antiaffinity rules which prohibit them to run on the same node, then podA will be evicted from the node so that podB and podC could run. This issue could happen, when the anti-affinity rules for pods B,C are created when they are already running on node. Currently, there are no parameters associated with this strategy. To disable this strategy, the policy should look like:
+This strategy makes sure that pods violating interpod anti-affinity are removed from nodes. For example, if there is podA on a node and podB and podC(running on the same node) have antiaffinity rules which prohibit them to run on the same node, then podA will be evicted from the node so that podB and podC could run. This issue could happen, when the anti-affinity rules for pods B,C are created when they are already running on node. Currently, there are no parameters associated with this strategy. To disable this strategy, the policy should look like this.
```
Comment (ingvagabund) [inline README.md]: s/podC(running/podC (running/

Comment (ingvagabund) [reply README.md]: Can you split the single line into multiple lines? The line is too long.

Comment (ingvagabund) [reply README.md]: `:` at the end is still more suitable here

File: README.md [hunk 7]
```diff
@@ -163,7 +164,7 @@ strategies:
 
 ### RemovePodsViolatingNodeAffinity
 
-This strategy makes sure that pods violating node affinity are removed from nodes. For example, there is podA that was scheduled on nodeA which satisfied the node affinity rule `requiredDuringSchedulingIgnoredDuringExecution` at the time of scheduling, but over time nodeA no longer satisfies the rule, then if another node nodeB is available that satisfies the node affinity rule, then podA will be evicted from nodeA. The policy file should like this -
+This strategy makes sure that pods violating node affinity are removed from nodes. For example, there is podA that was scheduled on nodeA which satisfied the node affinity rule `requiredDuringSchedulingIgnoredDuringExecution` at the time of scheduling, but over time nodeA no longer satisfies the rule, then if another node nodeB is available that satisfies the node affinity rule, then podA will be evicted from nodeA. The policy file should like this.
```
Comment (ingvagabund) [inline README.md]: Can you break the line into more lines?

File: README.md [hunk 8]
```diff
@@ -175,9 +176,10 @@ strategies:
       nodeAffinityType:
       - "requiredDuringSchedulingIgnoredDuringExecution"
 ```
+
 ### RemovePodsViolatingNodeTaints
 
-This strategy makes sure that pods violating NoSchedule taints on nodes are removed. For example: there is a pod "podA" with toleration to tolerate a taint ``key=value:NoSchedule`` scheduled and running on the tainted node. If the node's taint is subsequently updated/removed, taint is no longer satisfied by its pods' tolerations and will be evicted. The policy file should look like:
+This strategy makes sure that pods violating NoSchedule taints on nodes are removed. For example there is a pod "podA" with a toleration to tolerate a taint ``key=value:NoSchedule`` scheduled and running on the tainted node. If the node's taint is subsequently updated/removed, taint is no longer satisfied by its pods' tolerations and will be evicted. The policy file should look like this.
```
Comment (ingvagabund) [inline README.md]: `:` is still more suitable here
can you break the line into more lines?

File: README.md [hunk 9]
```diff
@@ -186,23 +188,25 @@ strategies:
   "RemovePodsViolatingNodeTaints":
     enabled: true
 ````
+
 ## Pod Evictions
 
-When the descheduler decides to evict pods from a node, it employs following general mechanism:
+When the descheduler decides to evict pods from a node, it employs the following general mechanism.
```
Comment (ingvagabund) [inline README.md]: `:` is still more suitable here

File: README.md [hunk 10]
```diff
@@ -186,23 +188,25 @@ strategies:
...
 never evicted because these pods won't be recreated.
 * Pods associated with DaemonSets are never evicted.
 * Pods with local storage are never evicted.
-* Best efforts pods are evicted before Burstable and Guaranteed pods.
-* All types of pods with annotation descheduler.alpha.kubernetes.io/evict are evicted. This 
-annotation is used to override checks which prevent eviction and user can select which pod is evicted. 
-User should know how and if the pod will be recreated.
+* Best efforts pods are evicted before burstable and guaranteed pods.
+* All types of pods with the annotation descheduler.alpha.kubernetes.io/evict are evicted. This 
+annotation is used to override checks which prevent eviction and users can select which pod is evicted.
```
Comment (ingvagabund) [inline README.md]: can you indent the line so it aligns with `All types of pods`?

File: README.md (hunk 1 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 5 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 6 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 7 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 8 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 9 above)
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md (hunk 10 above)
Comment (seanmalloy) [reply README.md]: Fixed

**Comment by seanmalloy** [comment]
@ingvagabund thanks for the review! I addressed all of your review comments. Please review again when you have some time.

File: README.md [hunk 11]
```diff
@@ -121,7 +121,12 @@ under utilized frequently or for a short period of time. By default, `numberOfNo
 
 ### RemovePodsViolatingInterPodAntiAffinity
 
-This strategy makes sure that pods violating interpod anti-affinity are removed from nodes. For example, if there is podA on a node and podB and podC(running on the same node) have antiaffinity rules which prohibit them to run on the same node, then podA will be evicted from the node so that podB and podC could run. This issue could happen, when the anti-affinity rules for pods B,C are created when they are already running on node. Currently, there are no parameters associated with this strategy. To disable this strategy, the policy should look like this.
+This strategy makes sure that pods violating interpod anti-affinity are removed from nodes. For example,
+if there is podA on a node and podB and podC (running on the same node) have anti-affinity rules which prohibit
+them to run on the same node, then podA will be evicted from the node so that podB and podC could run. This
+issue could happen, when the anti-affinity rules for pods B,C are created when they are already running on
```
[the remark this answers is not in the document]
Comment (seanmalloy) [reply README.md]: Fixed

File: README.md [hunk 12]
```diff
@@ -133,7 +138,11 @@ strategies:
 
 ### RemovePodsViolatingNodeAffinity
 
-This strategy makes sure that pods violating node affinity are removed from nodes. For example, there is podA that was scheduled on nodeA which satisfied the node affinity rule `requiredDuringSchedulingIgnoredDuringExecution` at the time of scheduling, but over time nodeA no longer satisfies the rule, then if another node nodeB is available that satisfies the node affinity rule, then podA will be evicted from nodeA. The policy file should like this.
+This strategy makes sure that pods violating node affinity are removed from nodes. For example, there is
+podA that was scheduled on nodeA which satisfied the node affinity rule `requiredDuringSchedulingIgnoredDuringExecution`
+at the time of scheduling, but over time nodeA no longer satisfies the rule, then if another node nodeB
+is available that satisfies the node affinity rule, then podA will be evicted from nodeA. The policy file
+should looks like:
```
[the remark this answers is not in the document]
Comment (seanmalloy) [reply README.md]: Fixed

**Comment by seanmalloy** [comment]
@ingvagabund I fixed that last two review comments. Let me know if you find anything else. Thanks!

**Pull request merged**